src/fixed.jl

```diff
@@ -16,27 +11,48 @@ function MP.polynomialtype(basis::AbstractPolynomialVectorBasis{PT}, T::Type) wh
 end
 function MP.polynomial(f::Function, basis::AbstractPolynomialVectorBasis)
     return MP.polynomial(mapreduce(
-        ip -> f(ip[1]) * ip[2], MA.add!, enumerate(basis.polynomials)))
+        ip -> f(ip[1]) * ip[2], MA.add!, enumerate(basis.elements)))
 end
 
 function MP.polynomial(Q::AbstractMatrix, basis::AbstractPolynomialVectorBasis,
                        T::Type)
     n = length(basis)
     @assert size(Q) == (n, n)
-    return MP.polynomial(mapreduce(row -> basis.polynomials[row] *
-        mapreduce(col -> Q[row, col] * basis.polynomials[col], MA.add!, 1:n),
+    return MP.polynomial(mapreduce(row -> basis.elements[row] *
+        mapreduce(col -> Q[row, col] * basis.elements[col], MA.add!, 1:n),
         MA.add!, 1:n), T)
 end
 
 """
     struct FixedPolynomialBasis{PT<:MP.AbstractPolynomialLike, PV<:AbstractVector{PT}} <: AbstractPolynomialBasis
-        polynomials::PV
+        elements::PV
     end
 
-Polynomial basis with the polynomials of the vector `polynomials`.
+Polynomial basis with the polynomials of the vector `elements`.
 For instance, `FixedPolynomialBasis([1, x, 2x^2-1, 4x^3-3x])` is the Chebyshev
 polynomial basis for cubic polynomials in the variable `x`.
 """
 struct FixedPolynomialBasis{PT<:MP.AbstractPolynomialLike, PV<:AbstractVector{PT}} <: AbstractPolynomialVectorBasis{PT, PV}
-    polynomials::PV
+    elements::PV
 end
+
+#=
+As `p` is an element of the vector space generated by `basis`, there exist unique coefficients `coeffs` such that `p = sum(coeffs[a]*a for a in basis)`. Now, when I take `b` in basis of highest degree, `c, rem = divrem(p, a)` should return `rem = sum(coeffs[a]*a for a in basis if !(a==b))` and `c = p[b]`, because `b` does not divide any of the other `a`. To see the last assertion, assume `b` divides `a`. Since `b` has maximal degree, `a` has maximal degree, too, and `a = bc` for some constant `c`. This however, contradicts that `p` has a unique representation `p = sum(coeffs[a]*a for a in basis)`, because
+```
+p[b]*b+ p[a]*a = p[b]*b + p[a]*b*c = (p[b]+ p[a]*c)*b + 0*a.
+```
+=#
+function MP.coefficients(p, Basis::Type{<:AbstractPolynomialBasis})
+    basis = maxdegree_basis(Basis, variables(p), maxdegree(p))
```

Comment:
Define
```julia
function MP.coefficients(p, Basis::Type{<:AbstractPolynomialBasis})
    return MP.coefficients(p, maxdegree_basis(Basis, variables(p), maxdegree(p)))
end
function MP.coefficients(p, basis::AbstractPolynomialBasis)
    # ...
end
```
It is needed for https://github.com/JuliaAlgebra/MultivariateMoments.jl/pull/16/files#r438604571

Reply:
Specifying an instantiated basis would kind of imply that we return a vector of coefficients of length `length(basis)`. So far I am deleting zero coefficients to be consistent with `coefficients(p)`.